components/card/card.js

```diff
@@ -0,0 +1,337 @@
+		this._tooltipShowing = false;
+		this._onBadgeResize = this._onBadgeResize.bind(this);
+		this._onFooterResize = this._onFooterResize.bind(this);
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		const badgeObserver = new ResizeObserver(this._onBadgeResize);
+		badgeObserver.observe(this.shadowRoot.querySelector('.d2l-card-badge'));
+		const footerObserver = new ResizeObserver(this._onFooterResize);
+		footerObserver.observe(this.shadowRoot.querySelector('.d2l-card-footer'));
+	}
+
+	render() {
+
+		const containerClass = {
+			'd2l-card-container': true,
+			'd2l-visible-on-ancestor-target': true,
+			'd2l-card-footer-hidden': this._footerHidden
+		};
+
+		const linkContainerClass = {
+			'd2l-card-link-container': true,
+			'd2l-card-link-container-hover': this._hover
+		};
+
+		const badgeStyle = {};
+		if (this._badgeMarginTop) badgeStyle.marginTop = this._badgeMarginTop;
+
+		const footerClass = {
+			'd2l-card-footer': true,
+			'd2l-offscreen': this._footerHidden
+		};
+
+		return html`
+			<div class="${classMap(containerClass)}"
+				@d2l-dropdown-open="${this._onDropdownOpen}"
+				@d2l-dropdown-close="${this._onDropdownClose}"
+				@d2l-tooltip-show="${this._onTooltipShow}"
+				@d2l-tooltip-hide="${this._onTooltipHide}">
+				<a @blur="${this._onLinkBlur}"
+					?download="${this.download}"
+					@focus="${this._onLinkFocus}"
+					href="${ifDefined(this.href)}"
+					hreflang="${ifDefined(this.hreflang)}"
+					@mouseenter="${this._onLinkMouseEnter}"
+					@mouseleave="${this._onLinkMouseLeave}"
+					rel="${ifDefined(this.rel)}"
+					target="${ifDefined(this.target)}"
+					type="${ifDefined(this.type)}">
+					<span class="d2l-card-link-text d2l-offscreen">${this.text}</span>
+				</a>
+				<div class="${classMap(linkContainerClass)}">
+					<div class="d2l-card-header"><slot name="header"></slot></div>
+					<div class="d2l-card-badge" style="${styleMap(badgeStyle)}"><slot name="badge"></slot></div>
+					<div class="d2l-card-content"><slot name="content"></slot></div>
+				</div>
+				<div class="d2l-card-actions"><slot name="actions"></slot></div>
+				<div class="${classMap(footerClass)}"><slot name="footer"></slot></div>
+			</div>
+		`;
+	}
+
+	focus() {
+		const elem = this.shadowRoot.querySelector('a');
+		if (!elem) return;
+		elem.focus();
+	}
+
+	_onBadgeResize(entries) {
+		if (!entries || entries.length === 0) return;
+		const entry = entries[0];
+		this._badgeMarginTop = `${-0.5 * entry.contentRect.height}px`;
+	}
+
+	_onDropdownClose() {
+		this._dropdownActionOpen = false;
+	}
+
+	_onDropdownOpen() {
+		this._dropdownActionOpen = true;
+	}
+
+	_onFooterResize(entries) {
+		if (!entries || entries.length === 0) return;
+		const entry = entries[0];
+		// firefox has a rounding error when calculating the height of the contentRect
+		// with `box-sizing: border-box;` so check for numbers which are close to 0 as well
+		this._footerHidden = (entry.contentRect.height < 1);
```

Comment:
I think the logic here has _technically_ changed, but is definitely worth the simplification.  I don't think we'll run into a situation where the footer was previous hidden, and then gets something 0.5 pixels high that we want to show.

Reply:
Cool yep... thanks for noticing that!  I should have called that out as something that changed.